### PR TITLE
add process exclusion list to clr profiler

### PR DESCRIPTION
**Pull request by lucaspimentel**
Changes proposed in this pull request:
- add a profiler exclusion list (by process name)
- change `devenv.bat` to use the new exclusion list so we don't need to keep adding every sample app ever
- bonus: replace useless log message about not being able to deserialize json (if the file is not found) with a message that includes the filename

### CODE REVIEW

File: src/Datadog.Trace.ClrProfiler.Native/integration_loader.cpp [hunk 1]
```diff
@@ -29,7 +29,13 @@ std::vector<Integration> LoadIntegrationsFromFile(const WSTRING& file_path) {
   try {
     std::ifstream stream;
     stream.open(ToString(file_path));
-    integrations = LoadIntegrationsFromStream(stream);
+
+    if (static_cast<bool>(stream)) {
```
Comment (lucaspimentel) [inline src/Datadog.Trace.ClrProfiler.Native/integration_loader.cpp]: This change is unrelated to the exclusion list. It replaces useless log message about not being able to deserialize json (if the file is not found) with a message that includes the filename.

File: src/Datadog.Trace.ClrProfiler.Native/integration_loader.cpp [hunk 2]
```diff
@@ -112,14 +118,18 @@ std::pair<MethodReplacement, bool> MethodReplacementFromJson(
     return std::make_pair<MethodReplacement, bool>({}, false);
   }
 
-  const auto caller = MethodReferenceFromJson(src.value("caller", json::object()), false);
-  const auto target = MethodReferenceFromJson(src.value("target", json::object()), true);
-  const auto wrapper = MethodReferenceFromJson(src.value("wrapper", json::object()), false);
+  const auto caller =
```
Comment (lucaspimentel) [inline src/Datadog.Trace.ClrProfiler.Native/integration_loader.cpp]: Reformatting only. Nothing to see from here down in this file.

File: src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp [hunk 3]
```diff
@@ -40,17 +40,32 @@ CorProfiler::Initialize(IUnknown* cor_profiler_info_unknown) {
...
-  // check if there is a whitelist of process names
-  const auto allowed_process_names =
-      GetEnvironmentValues(environment::process_names);
+  // check if there is a process name inclusion list
+  const auto include_process_names =
+      GetEnvironmentValues(environment::include_process_names);
 
-  if (!allowed_process_names.empty()) {
+  if (!include_process_names.empty()) {
     const auto process_name = GetCurrentProcessName();
```
Comment (colin-higgins) [inline src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp]: Would it save us much to consolidate the calls to `GetCurrentProcessName`?

Comment (lucaspimentel) [reply src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp]: Good catch. Refactored to only call once.
